examples/ImageRecon/GEOmetrics/utils.py

```diff
@@ -14,489 +14,485 @@
+
+        A = x2s - cur_xs
+        B = cur_xs - x1s
+        G = y2s - cur_ys
+        H = cur_ys - y1s
+
+        y1s = y1s + torch.arange(positions.shape[0]).float().to(device) * dim 
+        y2s = y2s + torch.arange(positions.shape[0]).float().to(device) * dim 
+
+        data_meta['A'].append(A.float().unsqueeze(0))
+        data_meta['B'].append(B.float().unsqueeze(0))
+        data_meta['G'].append(G.float().unsqueeze(0))
+        data_meta['H'].append(H.float().unsqueeze(0))
+        data_meta['x1s'].append(x1s.long().unsqueeze(0))
+        data_meta['x2s'].append(x2s.long().unsqueeze(0))
+        data_meta['y1s'].append(y1s.long().unsqueeze(0))
+        data_meta['y2s'].append(y2s.long().unsqueeze(0))
+
+    for key in data_meta:
+        data_meta[key] = torch.cat(data_meta[key], dim=0)
+    return data_meta
 
 
 def pooling(blocks, pooling_indices, index):
 
-	full_features = None 
-	for i_block, block in enumerate(blocks):
-		block = block[index]
-		A = pooling_indices['A'][i_block]
-		B = pooling_indices['B'][i_block]
-		G = pooling_indices['G'][i_block]
-		H = pooling_indices['H'][i_block]
+    full_features = None 
+    for i_block, block in enumerate(blocks):
+        block = block[index]
+        A = pooling_indices['A'][i_block]
+        B = pooling_indices['B'][i_block]
+        G = pooling_indices['G'][i_block]
+        H = pooling_indices['H'][i_block]
 
-		x1s = pooling_indices['x1s'][i_block]
-		x2s = pooling_indices['x2s'][i_block]
-		y1s = pooling_indices['y1s'][i_block]
-		y2s = pooling_indices['y2s'][i_block]
+        x1s = pooling_indices['x1s'][i_block]
+        x2s = pooling_indices['x2s'][i_block]
+        y1s = pooling_indices['y1s'][i_block]
+        y2s = pooling_indices['y2s'][i_block]
 
 
-		C =torch.index_select(block, 1, x1s).view(block.shape[0], -1 )
-		C = torch.index_select(C, 1, y1s)
-		D =torch.index_select(block, 1, x1s).view(block.shape[0], -1 )
-		D = torch.index_select(D, 1, y2s)
-		E =torch.index_select(block, 1, x2s).view(block.shape[0], -1 )
-		E = torch.index_select(E, 1, y1s)
-		F =torch.index_select(block, 1, x2s).view(block.shape[0], -1 )
-		F = torch.index_select(F, 1, y2s)
+        C = torch.index_select(block, 1, x1s).view(block.shape[0], -1)
+        C = torch.index_select(C, 1, y1s)
+        D = torch.index_select(block, 1, x1s).view(block.shape[0], -1)
+        D = torch.index_select(D, 1, y2s)
+        E = torch.index_select(block, 1, x2s).view(block.shape[0], -1)
+        E = torch.index_select(E, 1, y1s)
+        F = torch.index_select(block, 1, x2s).view(block.shape[0], -1)
+        F = torch.index_select(F, 1, y2s)
 
 
-		features = (A*C*G + H*D*A + G*E*B + B*F*H).permute(1,0)
+        features = (A*C*G + H*D*A + G*E*B + B*F*H).permute(1, 0)
 
-		if full_features is None: full_features = features
-		else: full_features = torch.cat((full_features, features), dim = 1)
- 
-	return full_features
+        if full_features is None:
```

Comment:
Not really important but `full_features = features if full_features if None else torch.cat((full_features, features), dim=1)` is more pythonic

Reply:
Done.